### DISCUSSION TITLE
Changing LID parameters during the simulation

### DISCUSSION THREAD

**Discussion by joshipra** [pyswmm/pyswmm#546] [Q&A] [answered]
I am aware that only the drainage parameters of the LIDs can be changed _**during**_ the simulation, but I was wondering if there is a workaround for changing other parameters as well (for example, soil porosity).

**Comment by karosc** [comment, marked as answer]
There is an example of this in the docs:
https://pyswmm.github.io/pyswmm/reference/api/pyswmm.lidlayers.Soil.html

Setting soil porosity during the simulation is not supported. This is because there are complications with adjusting LID structure definition when there is actively water in the LID. What happens to the water level when porosity is adjusted with water in the LID? Is there flooding? How do we handle that? How does all of this affect the mass balence? These are just some of the questions that need to be answered before implementing setters that can be called during a simulation. 

That being said, the feature is definitely possible to implement, you'd just need to understand how SWMM initializes the LID and tracks mass to make sure when the API makes it's change, the mass balance is maintained.  Whoever wrote the routine in the first place might have felt that setting soil porosity mid-simulation was more trouble than it was worth to them....or they didn't need the feature at all.

**Reply by joshipra** [reply]
Thank you for this. I am working on this with my colleague and have also been talking to Bryant E. McDonnell for his insights.
